BUGFIX: Asset related nodes listing

ComiR opened the pull request:
multiple fixes for the asset related node listing.
1. Internal nodedata will be skipped since they don't result in any nodes by themselves (i.e. nodes marked as deleted)
2. Changed linking of nearest document of asset related node:
   - only will be linked to if node in question is accessible (so the document is too)
   - linked directly to the user workspace ("open in backend") - the user can edit the node in question directly without searching it again in the backend
   - added translation if node is not accessible
   - new translation `edit.argument` with an argument (useful e.g. for `de` where the word order differs from `en`)
3. Only find nodes in the `sites` tree (`/sites/.*`) to prevent exceptions if other trees in the CR use the asset (exception 1430075362 would be thrown)

aertmann commented:
there's a pending pull request regarding the last issue #547, but could make sense to hotfix it for now until the other one is fixed

aertmann commented on TYPO3.Neos/Resources/Private/Templates/Media/Asset/RelatedNodes.html:
```diff
@@ -59,12 +59,12 @@
 								<f:for each="{documentNode.nodes}" as="relatedNode">
 									<li>
-										<f:if condition="{relatedNode.contextDocumentNode}">
+										<f:if condition="{relatedNode.accessible}">
```
changing this break the check if there is even a contextDocumentNode.. it probably needs to check for both actually, because either you don't have access or the node is missing..

ComiR replied:
If the `relatedNode` is accessible, the `contextDocumentNode` also must be accessible, or did I understand something wrong there?

If there is no `contextDocumentNode`, the `relatedNode` is an orphan and should be removed by `node:repair`.
I could handle that more gracefully. Will add that.

aertmann commented on TYPO3.Neos/Classes/TYPO3/Neos/Domain/Strategy/AssetUsageInNodePropertiesStrategy.php:
```diff
@@ -88,15 +90,19 @@ public function getUsageReferences(AssetInterface $asset)
+                $context = $this->createContentContext($userWorkspaceName);
             } else {
-                $context = $this->createContentContext($userWorkspace->getName());
+                $context = $this->createContextMatchingNodeData($relatedNodeData);
```
switching these around you're assuming you can actually access the node in your own workspace, pretty sure this will just lead to other problems..

ComiR replied:
You're right. I was assuming the own workspace is on top of the node workspace (i.e. user/live).
I'll change that.

aertmann commented:
Hey @ComiR thanks for the fixes, unfortunately I don't think they can go in like this..

ComiR commented:
I rebased onto the current `2.3` branch and changed the `isInternal()` check to `isRemoved()`.

`isInternal()` only checks if the node was moved. Since that means it is also removed (in the old path), I should be able to catch both cases with `isRemoved()`.

ComiR commented:
Now I don't even return those nodes in `getRelatedNodes()`...

ComiR commented:
#1486 did a fine job preventig the exception ;)
